### PR TITLE
 Push Command-> Goto Android Set Default App page (Android 7.0+) through Firebase

### PR DESCRIPTION
**Pull request by cnevinc**
#1983 
To sums up, this PR helps

1. Move all launching behaviour into LaunchIntentDispatcher.
2. Add the ability to handle commands from push notifications.

The next step will be:

1.  Make Rocket support intent scheme
2. Remove duplicated TelemetryEvent (MainActivity.sendBrowsingTelemetry())
3. Move other intent parsing work from MainActivity to LaunchIntentDispatcher

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/activity/TextActionActivity.java [hunk 1]
```diff
@@ -38,9 +38,9 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         }
         final String searchUrl = SearchUtils.createSearchUrl(this, searchText);
 
-        final Intent searchIntent = new Intent(this, MainActivity.class);
+        final Intent searchIntent = new Intent(this, RocketLauncherActivity.class);
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/focus/activity/TextActionActivity.java]: How about `searchIntent.setClassName(this, AppConstants.LAUNCHER_ACTIVITY_ALIAS);`?

File: app/src/main/java/org/mozilla/focus/notification/NotificationActionBroadcastReceiver.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public void onReceive(Context context, Intent intent) {
             NotificationManagerCompat.from(context).cancel(NotificationId.DEFAULT_BROWSER);
 
         } else if (bundle.getBoolean(IntentUtils.EXTRA_NOTIFICATION_CLICK_LOVE_FIREFOX)) {
-            nexStep = new Intent(context, MainActivity.class);
+            nexStep = new Intent(context, RocketLauncherActivity.class);
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/focus/notification/NotificationActionBroadcastReceiver.java]: How about `nexStep.setClassName(this, AppConstants.LAUNCHER_ACTIVITY_ALIAS);`?

File: app/src/main/java/org/mozilla/rocket/component/LaunchIntentDispatcher.kt [hunk 3]
```diff
@@ -32,19 +38,26 @@ class LaunchIntentDispatcher {
...
-                new.action = Intent.ACTION_VIEW
-                new.setClass(context, MainActivity::class.java)
-                new.putExtra(IntentUtils.EXTRA_OPEN_NEW_TAB, true)
-                context.startActivity(new)
-                return Action.HANDLED
+            intent.getStringExtra(RocketMessagingService.PUSH_OPEN_URL)?.run {
+
+                intent.data = Uri.parse(this)
+                intent.setClass(context, MainActivity::class.java)
+                intent.putExtra(IntentUtils.EXTRA_OPEN_NEW_TAB, true)
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/component/LaunchIntentDispatcher.kt]: `intent.action = Intent.ACTION_VIEW` is missing?

File: app/src/main/java/org/mozilla/focus/notification/RocketMessagingService.java [hunk 4]
```diff
@@ -7,29 +7,23 @@
...
 
     //
     @Override
-    public void onRemoteMessage(String url, String title, String body) {
+    public void onRemoteMessage(Intent intent, String title, String body) {
         if (!TelemetryWrapper.isTelemetryEnabled(this)) {
             return;
         }
+        // RocketLauncherActivity will handle this intent
+        intent.setClass(getApplicationContext(), RocketLauncherActivity.class);
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/focus/notification/RocketMessagingService.java]: How about `intent.setClassName(this, AppConstants.LAUNCHER_ACTIVITY_ALIAS);`?

File: app/src/main/java/org/mozilla/rocket/component/LaunchIntentDispatcher.kt (hunk 3 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/component/LaunchIntentDispatcher.kt]: yes!

File: app/src/main/java/org/mozilla/focus/notification/RocketMessagingService.java (hunk 4 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/notification/RocketMessagingService.java]: thx~!

**Pull request merged**